plugins/test/chromatic_correction_test.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright © 2024 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import unittest
+
+import numpy
+
+from odemis import model, dataio
+from odemis.dataio.tiff import read_data
+from odemis.util.spot import MaximaFind
+from odemis.util.transform import AffineTransform, alt_transformation_matrix_to_implicit, _rotation_matrix_from_angle
+from plugins.chromatic_correction import (BEAD_QUANTITY, SCALE, TOLERANCE,
+                                          find_corresponding_points, get_chromatic_correction_dict)
+
+PIXEL_SIZE = (1e-6, 1e-6)  # m
+# Affine parameters that are used on reference image
+# Translation can be any value. Scale is fixed to one if there is some rotation and shear else both scale, shear
+# can be changed by keeping the rotation to zero
+SHEAR = 0.9
+SCALE = [2, 3]
+ROTATION = numpy.radians(0)
```

Comment:
why test with a rotation of 0?

Reply:
I have added two sets of affine parameters, one with rotation 0 and one with non-zero such that the the mix of shear, rotation, scale keeps it affine.